WordPress/Classes/Extensions/URLComponents+Parsing.swift

```diff
@@ -0,0 +1,19 @@
+import Foundation
+
+
+// MARK: - URLComponents
+//
+extension URLComponents {
+
+    /// Attempts to parse the URL contained within a Path, with a given query. Returns nil on failure.
+    ///
+    static func parseURL(path: String, query: String) -> URL? {
```

Comment:
I find this one confusing, as it's trying to do two different things: 

-  Turn a string into a URL (that can be called `parse`)
- Append a given query to the URL (that's not parsing)

I think this one's needed because `URL.query` is read-only, but I'd suggest extending `URL` instead with a more generic way to transform a URL via its components:

```swift
extension URL {
    func componentMap(_ transform: (inout URLComponents) -> Void) -> URL? {
        guard var components = URLComponents(url: self, resolvingAgainstBaseURL: false) else {
            return nil
        }
        transform(&components)
        return components.url
    }

    func withQuery(_ query: String) -> URL? {
        return componentMap({ $0.query = query })
    }

    func withQueryItems(_ queryItems: [URLQueryItem]) -> URL? {
        return componentMap({ $0.queryItems = queryItems })
    }

}

let url = URL(string: "https://wordpress.com/")!
url.withQuery("a=1&b=2")
// => https://wordpress.com/?a=1&b=2
url.withQueryItems([URLQueryItem(name: "a", value: "1"), URLQueryItem(name: "b", value: "2")])
// => https://wordpress.com/?a=1&b=2
```

Reply:
Although this extension sounds good, we'd need to instantiate two URLComponents, and run multiple calls, to accomplish the same results. If it's oaky with you, i'd rather keeping this helper this way, since it's much simpler (and it's documented as well).